### PR TITLE
Add SimpleLink::unique_id function.

### PR DESCRIPTION
**Pull request by dhylands**
Plus some cargo fmt changes

### CODE REVIEW

File: cc3200-sys/src/simplelink.rs [hunk 1]
```diff
@@ -1,13 +1,14 @@
 use core::convert::TryFrom;
 use core::fmt;
 
-// Redo c_like_enum using try_from: https://gist.github.com/anonymous/92537fff7c6a3d1ab048fcd38bdf08b4
+// Redo c_like_enum using try_from:
+// https://gist.github.com/anonymous/92537fff7c6a3d1ab048fcd38bdf08b4
```
Comment (fabricedesre) [inline cc3200-sys/src/simplelink.rs]: nit: actually that was done!

File: cc3200-sys/src/simplelink.rs [hunk 2]
```diff
@@ -217,15 +221,16 @@ c_like_enum! {
...
-// as NetConfigSet for consistency.
+pub const SL_MAC_ADDR_LEN: usize = 6;
+
+// NetConfigGet only has an Id.
 c_like_enum! {
     NetConfigGet {
-        MacAddress = 0x0200,
-        Ipv4StaP2pClientGetInfo = 0x0301,
+        MacAddress = 0x02,
+        Ipv4StaP2pClientGetInfo = 0x03,
```
Comment (fabricedesre) [inline cc3200-sys/src/simplelink.rs:227]: hm, is that right?

Comment (dhylands) [reply cc3200-sys/src/simplelink.rs:227]: It matches with this:
https://github.com/fabricedesre/cc3200-rs/blob/master/cc3200-sys/sdk/simplelink/include/netcfg.h#L78-L79
and the get-mac-address works.

Comment (dhylands) [reply cc3200-sys/src/simplelink.rs:227]: I verified the retrieved mac address interdependently using a network scanner.

Comment (fabricedesre) [reply cc3200-sys/src/simplelink.rs:227]: Ok, I was worried it could have been a rustfmt bug.

File: src/simplelink.rs [hunk 3]
```diff
@@ -134,6 +151,28 @@ impl SimpleLink {
...
+            p
+        } else {
+            ptr::null_mut()
+        };
+        let mut result_len = result.len() as u8;
+        try_wlan!(sl_NetCfgGet(config as u8,
+                               config_opt_ptr,
+                               &mut result_len,
+                               result.as_mut_ptr()) as i16);
+        Ok(unsafe { slice::from_raw_parts_mut(result.as_mut_ptr(), result_len as usize) })
```
Comment (fabricedesre) [inline src/simplelink.rs]: Would Ok(result[0..result_len]) work and save us from an unsafe{} ?

Comment (dhylands) [reply src/simplelink.rs]: Almost. It looks like `Ok(&mut result[0..result_len as usize])` works fine.

**Comment by dhylands** [comment]
@fabricedesre I addressed the review comments.

**Comment by fabricedesre** [comment]
r+

**Pull request merged**